daffodil-core/src/test/scala/org/apache/daffodil/dsom/TestExternalVariables.scala

```diff
@@ -332,7 +332,7 @@ class TestExternalVariables {
     val dp1 = pf.onPath("/")
     val dp2 = pf.onPath("/").withExternalVariables(variables)
 
-    val outputter = new ScalaXMLInfosetOutputter()
+    val outputter = new ScalaXMLInfosetOutputter(dp2.daffodilConfig.xmlConversionControl)
```

Comment:
This is a good point. 

I only attached the config to the dp because i learned by trying it, that the xmlConversionControl object needed to get passed to exactly everywhere the dp was already being passed.  Because to run something, in the CLI, or in the TDML runner, you have to configure both the daffodil DP, AND at least one infoset inputter/outputter, which usually means one of the XML ones. So the two objects (DP and config info) just go together everywhere. 

But I will look into refactoring this so that the DP object doesn't carry the config at all. 

That will straighten out the NullInfosetInputter as well.